pkg/platform/runtime/setup/setup.go

```diff
@@ -502,6 +503,13 @@ func (s *Setup) fetchAndInstallArtifactsFromBuildPlan(installFunc artifactInstal
 				artifactsToInstall = append(artifactsToInstall, a.ArtifactID)
 			}
 		}
+
+		// We also caclulate the artifacts to be built which includes more than the runtime artifacts.
+		// This is used to determine if we need to show the "build in progress" screen.
+		buildtimeArtifacts, err = buildplan.BuildtimeArtifacts(buildResult.Build)
+		if err != nil {
+			return nil, nil, errs.Wrap(err, "Could not get buildtime artifacts")
+		}
```

Comment:
If the build is ready then we don't need to wait for a build and including build-time artifacts in the progress reporting is moot as it would instantly complete anyway. It only makes sense to do this if wen can identify them when a build is ongoing.

Reply:
Is that not what's happening here? Above on line 490 we ask if the build is ready. In this else block `BuildReady` was false so assume the build is in progress and ask for the buildtime artifacts for progress reporting.

Reply:
D'oh.. github's diff was hiding the `} else {`..